quota-service/src/modules/user/userService.js

```diff
@@ -0,0 +1,55 @@
+const _ = require('lodash');
+const client = require('../../databaseClient');
+const config = require('../../utils/config');
+const { getTimeDiff } = require('../../utils/helpers');
+
+class UserService {
+  getCollection() {
+    return client.db(config.dbName).collection(config.collectionName);
+  }
+
+  resetLimitQuota(key) {
+    this.getCollection().update({ key }, {
+      $set: {
+        consumedQuota: 0,
+        lastReset: Date.now(),
+      },
+    });
+  }
+
+  async setLimitQuota(key, newLimit) {
+    const { result } = await this.getCollection().updateOne({ key }, {
+      $set: {
+        quotaLimit: newLimit,
+      },
+    });
+    return _.isEqual(result.nModified, 1);
+  }
+
+  async getLimitQuota(key) {
+    const { quotaLimit } = await this.getCollection().findOne({ key });
+    return quotaLimit;
+  }
+
+  async getAvailableQuota(key) {
+    const data = await this.getCollection().findOne({ key });
+    const { quotaLimit, consumedQuota, lastReset } = data;
+    if (getTimeDiff(Date.now(), lastReset) >= 1) {
+      this.resetLimitQuota(key);
+      return quotaLimit;
+    }
+
+    return quotaLimit - consumedQuota;
+  }
+
+  async consumeQuota(key, consumed) {
+    const { result } = await this.getCollection().updateOne({ key }, {
+      $set: {
+        consumedQuota: consumed,
```

Comment:
We need to increase, not to set. Every time one of the APIs says "I consumed 10 quotas" we need to add it. If we just set it can cause issues if two services are running in parallel